wp-includes/sqlite/class-wp-sqlite-translator.php

```diff
@@ -772,7 +772,23 @@ public function get_return_value() {
 	 * @throws Exception If the query is not supported.
 	 */
 	private function execute_mysql_query( $query ) {
-		$tokens           = ( new WP_SQLite_Lexer( $query ) )->tokens;
+		$tokens = ( new WP_SQLite_Lexer( $query ) )->tokens;
+
+		// SQLite does not support CURRENT_TIMESTAMP() calls with parentheses.
+		// Since CURRENT_TIMESTAMP() can appear in most types of SQL queries,
+		// let's remove the parentheses globally before further processing.
+		foreach ( $tokens as $i => $token ) {
+			if ( WP_SQLite_Token::TYPE_KEYWORD === $token->type && 'CURRENT_TIMESTAMP' === $token->keyword ) {
+				$paren_open  = $tokens[ $i + 1 ] ?? null;
+				$paren_close = $tokens[ $i + 2 ] ?? null;
+				if ( WP_SQLite_Token::TYPE_OPERATOR === $paren_open->type && '(' === $paren_open->value
+					&& WP_SQLite_Token::TYPE_OPERATOR === $paren_close->type && ')' === $paren_close->value ) {
+					unset( $tokens[ $i + 1 ], $tokens[ $i + 2 ] );
+				}
+			}
+		}
+		$tokens = array_values( $tokens );
+
```

Comment:
This code works as expected, but this seems like the wrong place to put it.
Ideally, we should clean up the brackets in the Lexer, but I couldn't find a good place to do it there.

@adamziel @aristath do you have any suggestions on where to put this code?

Reply:
@bgrgicak I tried to move it here: https://github.com/WordPress/sqlite-database-integration/pull/151/commits/43b628d7c512c2d694182f5f0f0821b9f459d84a

Does that make sense?

Reply:
This definitely better but it still feels like a "random" piece of code in that function.

I will let @adamziel and @aristath decide on this one as I don't know enough about the codebase to make this decision.

Reply:
Unfortunately, I have the opposite feedback. This Lexer class is a direct port of the [PHPMyAdmin Lexer](https://github.com/phpmyadmin/sql-parser/blob/master/src/Lexer.php) and is meant to tokenize MySQL queries. parentheses are valid in MySQL and removing them in there sounds like an abstraction leak. Not that it matters with the AST Parser/Lexer explorations coming. I'm happy to get this change in as it is as I don't think we'll stick to the current lexer in the longer term, but if you were keen to relocate it then its original location in the Translator class seem fine to me.

Reply:
@adamziel If you prefer the variant from [the first commit](https://github.com/WordPress/sqlite-database-integration/pull/151/commits/4043e6d5b2c1d6b7eb12450226a9ba5c725391f0), I can just drop the last one and repush, and we can go with that for now. I guess if the new approach replaces the old one, it can be still valuable to add the test case, and we can have the functionality earlier.

Reply:
Yes please and agreed

Reply:
@adamziel Thanks! Repushed.